scripts/benchmark_inference.py

```diff
@@ -58,12 +61,55 @@
     default=2,
     help="Batch size of mock input",
 )
-
+parser.add_argument(
+    "--max_new_tokens",
+    type=int,
+    default=256,
+    help="Batch size of mock input",
+)
+parser.add_argument(
+    "--compile_mode",
+    type=str,
+    help="Mode for compilation",
+    default="default",
+    choices=["default", "reduce-overhead"],
+)
 parser.add_argument(
     "--distributed",
     action="store_true",
     help="This is a distributed job (multiple instances run with RANK+WORLD_SIZE)",
 )
+parser.add_argument(
+    "--check_correctness",
```

Comment:
I think we should check correctness by default (maybe `--skip_assertions`?)

Reply:
@nairbv In the case of pre-allocated cache, the cache is essentially a reference and the assertion would not hold anymore in the single pass (with use_cache=True and past_key_values=cache). How would we want to handle that? We do not run into this with end to end sequence generation

Reply:
@JRosenkranz I'm not sure what you mean. the assertions that exist before this PR all passed (except in cases where there was a bug). Using cache is supposed to result in the same output of generate (or of a single next-token, assuming we preserve the same state and arguments).